games-action/supertuxkart/supertuxkart-1.2.recipe

```diff
@@ -0,0 +1,102 @@
+SUMMARY="3D open-source kart racing game"
+DESCRIPTION="SuperTuxKart is a free 3D kart racing game. It aims towards fun for 
+players of all skill levels, with item boxes giving random items, nitro, drifting and 
+more. Realism is not the focus.
+
+STK offers an online multiplayer mode, a local multiplayer mode, and a single player \
+mode against AIs with both custom races and a story mode to complete to unlock new \
+karts and tracks. It features Grand Prix where the goal is to get the most points over \
+several races.
+
+The game also features additional game modes besides normal races : time trials, \
+follow-the-leader, soccer, capture-the-flag and two types of battle mode."
+HOMEPAGE="https://supertuxkart.net"
+COPYRIGHT="2020 SuperTuxKart Team"
+LICENSE="GNU GPL v3"
+REVISION="1"
+SOURCE_URI="https://github.com/supertuxkart/stk-code/releases/download/$portVersion/supertuxkart-$portVersion-src.tar.xz"
+CHECKSUM_SHA256="052edf0afdbeb99583fe8676fb0ab80ecb6103fb88b7540f858d1b5fa1297d37"
+SOURCE_FILENAME="supertuxkart-$portVersion-src.tar.xz"
+SOURCE_DIR="SuperTuxKart-$portVersion-src"
+
+ARCHITECTURES="!x86_gcc2 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+	supertuxkart$secondaryArchSuffix = $portVersion
+	app:SuperTuxKart$secondaryArchSuffix
+	"
+
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libcurl$secondaryArchSuffix
+	lib:libenet$secondaryArchSuffix
+	lib:libfreetype$secondaryArchSuffix
+	lib:libGL$secondaryArchSuffix
+	lib:libglu$secondaryArchSuffix
+	lib:libglew$secondaryArchSuffix
+	lib:libharfbuzz$secondaryArchSuffix
+	lib:libjpeg$secondaryArchSuffix
+	lib:libopenal$secondaryArchSuffix
+	lib:libogg$secondaryArchSuffix
+	lib:libpng$secondaryArchSuffix
+	lib:libsdl2_2.0$secondaryArchSuffix
+	lib:libsqlite3$secondaryArchSuffix
+	lib:libssl$secondaryArchSuffix
+	lib:libvorbis$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libcurl$secondaryArchSuffix
+	devel:libenet$secondaryArchSuffix
+	devel:libfreetype$secondaryArchSuffix
+	devel:libGL$secondaryArchSuffix
+	devel:libglu$secondaryArchSuffix
+	devel:libglew$secondaryArchSuffix
+	devel:libharfbuzz$secondaryArchSuffix
+	devel:libjpeg$secondaryArchSuffix
+	devel:libpng$secondaryArchSuffix
+	devel:libopenal$secondaryArchSuffix
+	devel:libogg$secondaryArchSuffix
+	devel:libsdl2$secondaryArchSuffix
+	devel:libsqlite3$secondaryArchSuffix
+	devel:libssl$secondaryArchSuffix
+	devel:libvorbis$secondaryArchSuffix
+	devel:libz$secondaryArchSuffix
+	"
+
+BUILD_PREREQUIRES="
+	cmd:cmake
+	cmd:gcc$secondaryArchSuffix
+	cmd:ld$secondaryArchSuffix
+	cmd:make
+	cmd:pkg_config$secondaryArchSuffix
+	"
+
+BUILD()
+{
+	mkdir -p build
+	cd build
+
+	cmake .. \
+		-DCMAKE_BUILD_TYPE=STKRelease \
+		-DCUSTOM_BINDIR="$appsDir/SuperTuxKart" \
+		-DSTK_INSTALL_BINARY_DIR=$binDir \
+		-DCMAKE_INSTALL_PREFIX=$prefix \
+		-DCMAKE_INSTALL_LIBDIR=$libDir \
+		-DBUILD_RECORDER=OFF \
+		-DBUILD_SHARED_LIBS=ON \
```

Comment:
Probably better to not build shared libraries: https://gitweb.gentoo.org/repo/gentoo.git/tree/games-action/supertuxkart/supertuxkart-1.2.ebuild#n81

Reply:
Works fine n Haiku...no static libs preferred in system lib AFAIK...

Reply:
Do you build in "strict" mode with haikuporter? Got some errors on the included libraries earlier not being declared/mentioned ...

Reply:
Originally, yes. I need to re-review current recipe... had to hack things a bit to get a decent Haiku packaging layout.
Main shared libs are angelscript/squish. If you 'LD_PRELOAD' them manually, the game boots properly. The packaging info
I'm trying to fix for menu/deskbar and HDS for game icon display in menu/deskbar and changelog appearance (HDS). Game works fine on Haiku with shared libs and current dataDir/SuperTuxKart setup from packaging (may need some further tweaks).. I have a x86_64 package in my test repo which works with Haiku GUI (without any preloading) and will test the newer recipe on x86 today.

Reply:
angelscript can be disabled afaik, squish is another thing, haven't investigated that far (yet) :)

Reply:
Ok... I added them for now.  Will focus on any further packaging issues.

Reply:
OK, doing my end here on the 32bit side :)